### PR TITLE
Add code to check if upload status is 'Completed' & page in the case …

### PR DESCRIPTION
**Pull request by techncl**
…it is

### CODE REVIEW

File: app/validation/ValidatedActions.scala [hunk 1]
```diff
@@ -26,22 +26,24 @@ abstract class ValidatedActions() extends TokenSecurity with I18nSupport {
...
-      } else if(uploadInProgress) {
-        Ok(views.html.uploadInProgress(consignmentId))
-      } else {
-        f(request)
+        if (!transferAgreementExists) {
+          Redirect(routes.TransferAgreementController.transferAgreement(consignmentId))
+        } else if (uploadStatus == "InProgress") {
+          Ok(views.html.uploadInProgress(consignmentId))
+        } else if (uploadStatus == "Completed") {
+          Ok(views.html.uploadHasCompleted(consignmentId))
```
Comment (suzannehamilton) [inline app/validation/ValidatedActions.scala:43]: Could you add a controller test for this case?

File: app/views/uploadHasCompleted.scala.html [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+@(consignmentId: UUID)(implicit request: RequestHeader, messages: Messages)
+
+@main("Upload your records") {
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      @progressIndicator(Messages("upload.progress"))
+      <h1 class="govuk-heading-l">Uploading records</h1>
+
+      <div class="govuk-grid-column-two-thirds">
+        <p>Your upload is complete and has been saved. You cannot make amendments to your upload or add additional files</p>
```
Comment (suzannehamilton) [inline app/views/uploadHasCompleted.scala.html:11]: Should there be a full stop at the end of the second sentence?

File: app/views/uploadHasCompleted.scala.html [hunk 3]
```diff
@@ -0,0 +1,19 @@
+@import java.util.UUID
+@(consignmentId: UUID)(implicit request: RequestHeader, messages: Messages)
+
+@main("Upload your records") {
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+      @progressIndicator(Messages("upload.progress"))
```
Comment (suzannehamilton) [inline app/views/uploadHasCompleted.scala.html:7]: Should the progress bar be here? I wouldn't have expected to see it here, but I'm not sure of the design.

**Review by suzannehamilton — COMMENTED** [review]
Just a few comments, thanks!

File: app/views/uploadHasCompleted.scala.html (hunk 3 above)
Comment (techncl) [reply app/views/uploadHasCompleted.scala.html:7]: Oh no, this is calling the [progress indicator ](https://github.com/nationalarchives/tdr-transfer-frontend/blob/master/app/views/progressIndicator.scala.html) file, the one that shows which step you are on out of N steps

Comment (suzannehamilton) [reply app/views/uploadHasCompleted.scala.html:7]: You're quite right. Ignore me. 🙂

**Review by suzannehamilton — APPROVED** [review]
🚀

**Pull request merged**